mm2src/coins/utxo/utxo_block_header_storage/mod.rs

```diff
@@ -95,3 +95,216 @@ impl BlockHeaderStorageOps for BlockHeaderStorage {
         self.inner.get_block_height_by_hash(hash).await
     }
 }
+
+#[cfg(test)]
+mod block_headers_storage_tests {
+    cfg_wasm32! {
+        use super::*;
+        use super::IDBBlockHeadersStorage;
+        use chain::BlockHeaderBits;
+        use common::log::wasm_log::register_wasm_log;
+        use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+        use wasm_bindgen_test::*;
+        use spv_validation::work::MAX_BITS_BTC;
+
+        wasm_bindgen_test_configure!(run_in_browser);
+    }
+
+    cfg_native! {
+        use super::*;
+        use spv_validation::work::MAX_BITS_BTC;
+        use chain::BlockHeaderBits;
+        use mm2_test_helpers::for_tests::mm_ctx_with_custom_db;
+
+        use crate::utxo::utxo_block_header_storage::sql_block_header_storage::block_headers_cache_table;
+    }
+
+    #[cfg(target_arch = "wasm32")]
+    pub(crate) async fn block_headers_storage_for_test(_for_coin: &str) -> (MmArc, IDBBlockHeadersStorage, String) {
+        let ctx = mm_ctx_with_custom_db();
+        let storage = IDBBlockHeadersStorage::new(&ctx, "RICK".to_string());
+        register_wasm_log();
+
+        (ctx, storage, "".to_string())
+    }
+
+    #[cfg(not(target_arch = "wasm32"))]
+    pub(crate) async fn block_headers_storage_for_test(for_coin: &str) -> (MmArc, SqliteBlockHeadersStorage, String) {
+        let ctx = mm_ctx_with_custom_db();
+        let storage = SqliteBlockHeadersStorage::in_memory(for_coin.into());
+        let table = block_headers_cache_table(for_coin);
```

Comment:
Please consider refactoring this as:
```rust
let ctx = mm_ctx_with_custom_db();
let storage /*impl BlockHeaderStorageOps*/ = BlockHeadersStorageBuilder::build(ctx, for_coin);
```

Reply:
More details:
`mm_ctx_with_custom_db()` returns a `MmCtx` with
1) In-memory sqlite database if `not(target_arch = "wasm32")`
2) IndexedDb that points to a test namespace, so the databases between tests are not overlay.

So you can use the `MmCtx` to build a Block Headers Storage independently on the target arch.
For example,
```rust
// utxo_block_header_storage/mod.rs

struct BlockHeaderStorageBuilder {
  mm_ctx: MmArc,
  ticker: String,
}

impl BlockHeaderStorageBuilder {
  pub fn new(mm_ctx: MmArc, ticker: String) -> Self { todo!() }

  #[cfg(target_arch = "wasm32")]
  pub fn build(self) -> impl BlockHeaderStorage { IDBBlockHeadersStorage::new(self.mm_ctx, self.ticker) }

  // Similar for native targets
}
```

Then you could avoid adding the `block_headers_storage_for_test` function and do something like:
```rust
pub(crate) async fn test_add_block_headers_impl() {
        let ctx = mm_ctx_with_custom_db();
        let for_coin = "insert";
        let storage = BlockHeaderStorageBuilder::new(ctx.clone(), for_coin.to_string).build();

        let mut headers = HashMap::with_capacity(1);
        let block_header: BlockHeader = "0000002076d41d3e4b0bfd4c0d3b30aa69fdff3ed35d85829efd04000000000000000000b386498b583390959d9bac72346986e3015e83ac0b54bc7747a11a494ac35c94bb3ce65a53fb45177f7e311c".into();
        headers.insert(520481, block_header);
        storage.add_block_headers_to_storage(headers).await.unwrap();
        assert!(!storage.is_table_empty(&table).await);
    }
```

Reply:
I don't think this is needed since we already have 
https://github.com/KomodoPlatform/atomicDEX-API/blob/ba34d1702f71b959772e528a75ca98035ba892bf/mm2src/coins/utxo/utxo_block_header_storage/mod.rs#L64-L104

Reply:
I don't think this is needed since we already have
https://github.com/KomodoPlatform/atomicDEX-API/blob/ba34d1702f71b959772e528a75ca98035ba892bf/mm2src/coins/utxo/utxo_block_header_storage/mod.rs#L27-L60